manticore/native/memory.py

```diff
@@ -1320,14 +1320,14 @@ def _try_get_solutions(self, address, size, access, max_solutions=0x1000, force=
         """
         assert issymbolic(address)
         solver = Z3Solver.instance()
-        solutions = solver.get_all_values(self.constraints, address, maxcnt=max_solutions)
+        solutions = solver.get_all_values(self.constraints, address, maxcnt=max_solutions, silent=True)
 
         crashing_condition = False
         for base in solutions:
             if not self.access_ok(slice(base, base + size), access, force):
                 crashing_condition = Operators.OR(address == base, crashing_condition)
 
-        crash_or_not = solver.get_all_values(self.constraints, crashing_condition, maxcnt=3)
+        crash_or_not = solver.get_all_values(self.constraints, crashing_condition, maxcnt=2, silent=True)
```

Comment:
Still confused about what's going on here. We want either exactly one or exactly two solutions from `get_all_values`, and given that it checks if `len(results) >= maxcnt`, it seems like it should be three in this scenario, with `silent` set to `False`. Since this is a native change and this is an EVM PR, perhaps it would be worth extracting this so we can test it in isolation? Let's discuss on Slack tomorrow.

Reply:
`crashing_condition` being a Bool can have at the most 2 solutions. 
It's an optimization that avoids the third check-sat

crashing_condition != True
crashing_condition != False
check-sat
obviously unsat